starter_kits/Go/src/hlt/Player.go

```diff
@@ -27,6 +28,16 @@ func NewPlayer() *Player {
 	return &Player{playerID, NewShipyard(playerID, &Position{x, y}), 0, nil, nil}
 }
 
+// GetShip - Get a specific Ship by its ID.
+func (p *Player) GetShip(shipId int) (*Ship, error) {
```

Comment:
Would it be possible to somehow make it better than a linear search? Maybe if there is also a lookup map like in Java/C++? I'd be ok with binary search too.

Reply:
Great catch. Switching from linear search to a direct lookup, however, I'm not certain this won't break something. Specifically, it assumes the Player.Ships map[int]*Ship uses a non-changing ship ID as its map key, which matches the Entity.id assigned when the Ship was initially created.

Trying to verify this assumption myself:

Player.Update is called each turn, which recreates the Player.Ships map each turn, which instantiates new Ship instances via NewShip(player.ID) each turn. ref: https://github.com/HaliteChallenge/Halite-III/blob/master/starter_kits/Go/src/hlt/Player.go#L31
NewShip(player.ID) assigns ship.ID from input.GetInt() - ref: https://github.com/HaliteChallenge/Halite-III/blob/master/starter_kits/Go/src/hlt/Entities.go#L45
input.GetInt() scans for the ID from a string buffer which I don't see in the code, but which is likely the Ship's ID based on my experience implementing a Python bot... - ref: https://github.com/HaliteChallenge/Halite-III/blob/master/starter_kits/Go/src/hlt/input/Input.go
It'd be a good idea for someone more familiar with the expected string buffer to confirm before merging.

Reply:
Looks good to me as well!